src/server.c

```diff
@@ -4657,11 +4667,20 @@ sds genRedisInfoString(const char *section) {
             );
 
             if (server.repl_state == REPL_STATE_TRANSFER) {
+                double perc = 0;
+                if (server.repl_transfer_size) {
+                    perc = ((double)server.repl_transfer_read / server.repl_transfer_size) * 100;
+                }
                 info = sdscatprintf(info,
+                    "master_sync_total_bytes:%lld\r\n"
+                    "master_sync_read_bytes:%lld\r\n"
                     "master_sync_left_bytes:%lld\r\n"
-                    "master_sync_last_io_seconds_ago:%d\r\n"
-                    , (long long)
-                        (server.repl_transfer_size - server.repl_transfer_read),
+                    "master_sync_perc:%.2f\r\n"
+                    "master_sync_last_io_seconds_ago:%d\r\n",
+                    (long long) server.repl_transfer_size,
+                    (long long) server.repl_transfer_read,
+                    (long long) (server.repl_transfer_size - server.repl_transfer_read),
```

Comment:
I'm not sure if this is a problem, the result will be negative if master enable diskless replication(not your PR,), and `master_sync_total_bytes ` will be zero.

Reply:
yes, i know, this was already broken, but i also consider that entire field to be a bad idea, so i don't think i wanna fix it (just hope people will not/stop use it).
At least with the new fields, the caller can notice the size is zero and reach whatever conclusion he needs from that fact.

Note that in this case (master is diskless, and the repliac is disk-based) none of these fields are gonna provide any indication of progress (since we don't even have loading_rdb_used_mem). the only way around that is maybe have the replica send INFO MEMORY before it sends PSYNC.

Reply:
Your PR provides `master_sync_total_bytes` and `master_sync_read_bytes`, these are enough for users to judge even if `master_sync_left_bytes` is negative.

 For me, I think it is a good idea to show approximate progress without much complexity when diskless load

Reply:
Did you mean to suggest any change? 
I think I already show an approximate easy to use progress (best I can, 
And I can't trim master_sync_left_bytes.

Reply:
No suggestion, it is ok now.